### DISCUSSION TITLE
Account Deletion

### DISCUSSION THREAD

**Discussion by scottsignal** [apache/cloudstack#10128] [Ideas]
One of my colleagues had submitted https://github.com/apache/cloudstack/issues/9480.

We have discussed this internally a bit more since then and wondering if there are additional safeguard that could be in place. Rather than me submitting another enhancement I thought maybe it would be better to discuss it.

One of the ideas we came up with was that instances need to be powered off to delete an account.  

Kinda wondering what other thoughts are out there on this.  The concern here is really being able to destroy your entire environment if you don't understand as an end user that instances are tied to accounts.

**Comment by abh1sar** [comment]
hi @scottsignal, I am planning to work on #9480. 
I agree with the problem and like the ideas.

My suggestion is
1. Account has to be disabled first for it to be deleted (Disabling actually shuts down all the VMs).
2. Improve the warning message to say that all VMs, volumes and data associated with the account will be deleted.
3. Additional prompt to type the name of the account to confirm account deletion. 

Please let me know your thoughts.

**Reply by scottsignal** [reply]
@abh1sar Thank you for working on this.   This works and I like the idea of requiring the account to be disabled.
